registry/storage/driver/gcs/gcs.go

```diff
@@ -185,6 +195,11 @@ func FromParameters(parameters map[string]interface{}) (storagedriver.StorageDri
 			return nil, err
 		}
 		ts = jwtConf.TokenSource(context.Background())
+
+		gcs, err = storage.NewClient(context.Background(), option.WithCredentialsJSON(data))
```

Comment:
when do we have chance to call client.Close()?

Reply:
When driver is deregistered would be one.

Reply:
Here
https://github.dev/jmontleon/distribution/blob/65985dab284985512ea0c4b0ba16c8eff1f03c73/registry/storage/driver/gcs/gcs.go#L395

Reply:
I checked s3, no closing of the session client either.

Reply:
Does not need to be called if program is using GCS client throughout its lifetime till it exits per https://github.com/googleapis/google-cloud-go/blob/storage/v1.25.0/storage/storage.go#L233

Reply:
Which I believe registry use the client pretty much all the time to healthcheck the driver. So gcs.Close() need not apply.